### PR TITLE
front: migrate delete project in rtk

### LINKED ISSUE #5186: front: migrate last endpoint rtk
### Description and goal

One sneaky delete endpoint in AddOrEditProjectModal dodged the rtk migration, it should be using rtk.

### Acceptance criteria

- [ ] DeleteProject should use rtk

### PR DESCRIPTION
**Pull request by SharglutDev**
close #5186

### CODE REVIEW

File: front/src/modules/project/components/AddOrEditProjectModal.tsx [hunk 1]
```diff
@@ -156,22 +157,29 @@ export default function AddOrEditProjectModal({
...
-        dispatch(
-          setSuccess({
-            title: t('projectDeleted'),
-            text: t('projectDeletedDetails', { name: project.name }),
-          })
-        );
-      } catch (error) {
-        console.error(error);
-      }
+      await deleteProject({ projectId: 50 })
```
Comment (anisometropie) [inline front/src/modules/project/components/AddOrEditProjectModal.tsx]:
```suggestion
      await deleteProject({ projectId: project.id })
```

**Review by anisometropie — CHANGES_REQUESTED** [review]

File: front/src/modules/project/components/AddOrEditProjectModal.tsx (hunk 1 above)
Comment (SharglutDev) [reply front/src/modules/project/components/AddOrEditProjectModal.tsx]: Wow good catch, I forgot to change it back 😬

Comment (SharglutDev) [reply front/src/modules/project/components/AddOrEditProjectModal.tsx]: It's fixed

**Review by Uriel-Sautron — APPROVED** [review]
Good job !!!

**Review by anisometropie — APPROVED** [review]

**Review by clarani — APPROVED** [review]
LGTM, tested :heavy_check_mark:

**Pull request merged**